### DISCUSSION TITLE
Diffraction efficiency

### DISCUSSION THREAD

**Discussion by leok9547** [kc-ml2/meent#82] [General]
Hi, 

As I’m new to Meent, could someone please advise on finding the +1th or +2th or even higher order diffraction efficiency of reflection from TM component? Any hints or example code would be greatly appreciated.

Many thanks~

**Comment by yonghakim** [comment]
Hi and welcome,

https://github.com/kc-ml2/meent/blob/main/tutorials/01-modeling-and-emsolver.ipynb
You can see this notebook.

**Reply by leok9547** [reply]
Thanks for the reply. 
Actually, I did run the code in modeling-and-emsolver.ipynb. I use ucell_1d_s as ucell and fto = 100, it gave me a (1,201) array of de_ri.
But I cant figure out what is the meaning of value in the array.

Thanks~

**Reply by yonghakim** [reply]
Oh, I see.

The result is a 2D array that the first axis is the diffraction on Y-axis and the 2nd axis is for X-axis. 
In meent, X-axis is the default axis hence you get (1, 201) size which means there is no diffraction in Y-axis (only 0 mode) and 201 modes in X-axis. `fto` is the number of harmonics in use, so you will get all the modes from -fto to +fto including 0 mode.

So in your case, the result array is the diffraction efficiencies of the modes in the order of
-200, -199, ..., -1, 0, +1, ...,+199, +200
The 0th mode is at the center and +1th mode is the next one.

**Reply by leok9547** [reply]
Thank you for your kind help,
I got it now
